src/modules/Core/Wallets/api.js

```diff
@@ -82,21 +82,33 @@ export const getEnabledTokensFile = (wallet: AbcCurrencyWallet) => {
 }
 
 export async function setEnabledTokens (wallet: AbcCurrencyWallet, tokens: Array<string>, tokensToDisable?: Array<string>) {  // initialize array for eventual setting of file
-  let finalTextArray = tokens
+  let finalTextArray = [...tokens]
   // now stringify the new tokens
   let stringifiedTokens = JSON.stringify(finalTextArray)
   // grab the enabledTokensFile
+  const tokensFile = getEnabledTokensFile(wallet)
+  await tokensFile.setText(stringifiedTokens)
+  enableTokens(wallet, tokens)
+  if (tokensToDisable && tokensToDisable.length > 0) {
+    disableTokens(wallet, tokensToDisable)
+  }
+  return tokens
+}
+
+export async function updateEnabledTokens (wallet: AbcCurrencyWallet, tokensToEnable: Array<string>, tokensToDisable: Array<string>) {
```

Comment:
What is the purpose of having both setEnabledTokens and updateEnabledTokens? They seem to do almost exactly the same thing. You only use updateEnabledTokens for deleting a token, so shouldn't it just be deleteTokens() and you can remove the enable functionality

Reply:
We created the function as a more general way to enable or disable tokens, all in one function. So the idea would be to eventually get rid of `setEnabledTokens`. I can replace the `setEnabledTokens` calls with `updateEnabledTokens` as part of this pull request, if you'd like.